cmd/gke-certificates-controller/app/gke_approver.go

```diff
@@ -372,3 +392,153 @@ func loadApproverOptions(s *GKECertificatesController) (approverOptions, error)
 
 	return a, nil
 }
+
+var tpmAttestationBlocks = []string{
+	"CERTIFICATE REQUEST",
+	"VM IDENTITY",
+	"ATTESTATION DATA",
+	"ATTESTATION SIGNATURE",
+}
+
+func isNodeClientCertWithAttestation(opts approverOptions, csr *capi.CertificateSigningRequest, x509cr *x509.CertificateRequest) bool {
+	if !isNodeClientCert(opts, csr, x509cr) {
+		return false
+	}
+	blocks, err := parsePEMBlocks(csr.Spec.Request)
+	if err != nil {
+		glog.Errorf("parsing csr.Spec.Request: %v", err)
+		return false
+	}
+	for _, name := range tpmAttestationBlocks {
+		if _, ok := blocks[name]; !ok {
+			return false
+		}
+	}
+	return true
+}
+
+func validateTPMAttestation(opts approverOptions, csr *capi.CertificateSigningRequest, x509cr *x509.CertificateRequest) bool {
+	glog.Error("Validation of CSRs with TPM attestation is WIP")
+	return false
+
+	blocks, err := parsePEMBlocks(csr.Spec.Request)
+	if err != nil {
+		glog.Errorf("parsing csr.Spec.Request: %v", err)
+		return false
+	}
+	nodeIDRaw := blocks["VM IDENTITY"].Bytes
```

Comment:
How are we going to migrate to using EKCerts?

Reply:
We will branch this logic based on what blocks are present in CSR object:
- If we have `VM IDENTITY`, use public key from GCE API
- if we have `ENDORSEMENT CERTIFICATE`, use CA chain validation